python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -1,200 +1,265 @@
-    tags: list[str] = field(default_factory=list)
-    major_version: int = 1
-    row_count: int | None = None
-    owner: str = ""
-    owner_email: str = ""
-    last_updated: datetime | None = None
-    first_created: datetime | None = None
-
-    @staticmethod
-    def from_data_product_schema_dict(
-        metadata: dict[str, Any], table_name, retention_period: int | None = None
-    ):
-        """
-        Expects a dict containing data product table schema information as per the
-        required fields in the json schema at
-        https://github.com/ministryofjustice/modernisation-platform-environments/tree/main/terraform/environments/data-platform/data-product-table-schema-json-schema
-
-        and should be passed table name and optionally a retention period
-
-        Then populates a TableMetadata object with the given data.
-        """
-
-        new_metadata = TableMetadata(
-            name=table_name,
-            description=metadata["tableDescription"],
-            column_details=metadata["columns"],
-            retention_period_in_days=retention_period,
-            source_dataset_name=metadata.get("sourceDatasetName", ""),
-            where_to_access_dataset=metadata.get("sourceDatasetLocation", ""),
-            data_sensitivity_level=SecurityClassification[
-                metadata.get("securityClassification", "OFFICIAL")
-            ],
-            tags=metadata.get("tags", []),
-        )
-
-        return new_metadata
-
-
-# jscpd:ignore-start
-@dataclass
-class DatabaseMetadata:
-    """
-    For source system databases - currently matches DataProductMetadata
-    but will need to be revisted and refined potentially
-    """
-
-    name: str
-    description: str
-    version: str
-    owner: str
-    owner_display_name: str
-    maintainer: str | None
-    maintainer_display_name: str | None
-    email: str
-    retention_period_in_days: int
-    domain: str
-    subdomain: str | None
-    dpia_required: bool
-    dpia_location: str | None
-    last_updated: datetime
-    creation_date: datetime
-    s3_location: str | None
-    status: DataProductStatus = DataProductStatus.DRAFT
-    tags: list[str] = field(default_factory=list)
-
-
-# jscpd:ignore-end
-
-
-@dataclass
-class ChartMetadata:
-    name: str
-    description: str
-    external_url: str
-    tags: list[str] = field(default_factory=list)
+class Column(BaseModel):
+    """
+    A column definition in a table
+    """
+
+    name: str = Field(
+        pattern=r"^[\w.]+$",
+        description="The name of a column as it appears in the table.",
+        json_schema_extra={"pattern": r"^[\w.]+$"},
+    )
+    display_name: str = Field(description="A user-friendly version of the name")
+    type: str = Field(
+        description="The data type of the column as it appears in the table",
+    )
+    description: str = Field(description="A description of the column")
```

Comment:
We should come up with a way to clarify for all `description` properties whether we're referencing the `source` or `editable` schemaMetadataProperty, (see [python implementation for column description](https://datahubproject.io/docs/api/tutorials/descriptions/#add-description-on-dataset))

Reply:
This is going to be important really soon as we develop the manual registration. Shall we create a separate ticket for this?

Reply:
this should help inform https://github.com/ministryofjustice/find-moj-data/issues/287